index.js

```diff
@@ -43,63 +91,139 @@ NeoCities.prototype.get = function(method, args, callback) {
 
 NeoCities.prototype.info = function(sitename, callback) {
   var args = null
-
+  
   if(typeof sitename == 'function')
     callback = sitename
   else if(typeof sitename == 'string')
     args = {sitename: sitename}
-
+  
   this.get('info', args, callback)
 }
 
+NeoCities.prototype.list = function(path, callback) {
+  var args = null
+  
+  if (typeof path == 'function')
+    callback = path
+  else if (typeof path == 'string')
+    args = {path: path}
+  
+  this.get('list', args, callback)
+}
+
 NeoCities.prototype.post = function(method, args, callback) {
   var form = new FormData()
-  var i
-
-  for(i=0;i<args.length;i++)
+  
+  for(var i = 0; i < args.length; i ++)
     form.append(args[i].name, args[i].value)
-
-  var request = this.client.request({
+  
+  var opts = {
     method: 'post',
     host: this.url.hostname,
     port: this.url.port,
-    path: '/api/'+method,
-    headers: form.getHeaders(),
-    auth: this.user+':'+this.pass
-  }, function(res) {
+    path: '/api/' + method,
+    headers: form.getHeaders()
+  }
+  
+  if (!this.key) {
+    opts.auth = this.user + ':' + this.pass
+  }
+  else {
+    opts.headers.Authorization = 'Bearer ' + this.key
+  }
+  
+  var request = this.client.request(opts, function(res) {
     var body = ''
-
+    
     res.on('data', function (chunk) {
       body += chunk
     })
-
+    
     res.on('end', function() {
       var resObj = JSON.parse(body)
       callback(resObj)
     })
   })
-
+  
   form.pipe(request)
 }
 
 NeoCities.prototype.delete = function(filenames, callback) {
   var args = []
-  var i
-
-  for(i=0;i<filenames.length;i++)
+  
+  for(var i = 0; i < filenames.length; i ++)
     args.push({name: 'filenames[]', value: filenames[i]})
-
+  
   this.post('delete', args, callback)
 }
 
 NeoCities.prototype.upload = function(files, callback) {
   var args = []
-  var i
-
-  for(i=0;i<files.length;i++)
+  
+  for(var i = 0; i < files.length; i ++)
     args.push({name: files[i].name, value: fs.createReadStream(files[i].path)})
-
+  
   this.post('upload', args, callback)
 }
 
+NeoCities.prototype.push = function(localPath, webPath, excludes, callback) {
+  if (typeof webPath == 'function' || typeof webPath == 'object') {
+    callback = excludes
+    excludes = webPath
+    webPath = '/'
+  }
+  if (typeof excludes == 'function') {
+    callback = excludes
+    excludes = []
+  }
+  var activePaths = []
+  var that = this
+  excludes = excludes.map(function(dir) {return path.resolve(localPath, dir)})
+  list(localPath)
+  
+  function list(dir) {
+    activePaths.push(dir)
+    fs.readdir(dir, parseFiles)
+    function parseFiles(err, dirContents) {
```

Comment:
This would be cleaner and easier for others to understand if it had better logical separation

Reply:
I tried to make this better, but I don't have to worry about the readability of my code very often so I have no clue how well I did.